core/auth/src/main/java/software/amazon/awssdk/auth/token/credentials/SdkTokenProviderChain.java

```diff
@@ -79,21 +81,33 @@ public static SdkTokenProviderChain of(SdkTokenProvider... sdkTokenProviders) {
         return builder().tokenProviders(sdkTokenProviders).build();
     }
 
+    /**
+     * Create a token provider chain with default configuration that checks the given token providers.
+     * @param sdkTokenProviders The token providers that should be checked for token, in the order they should
+     *                                be checked.
+     * @return A token provider chain that checks the provided token providers in order.
+     */
+    public static SdkTokenProviderChain of(IdentityProvider<? extends TokenIdentity>... sdkTokenProviders) {
+        return builder().tokenProviders(sdkTokenProviders).build();
+    }
+
     @Override
     public SdkToken resolveToken() {
         if (reuseLastProviderEnabled && lastUsedProvider != null) {
-            return lastUsedProvider.resolveToken();
+            // TODO: Exception handling for join?
```

Comment:
Do you have a special format for TODOs? Can't remember how we're tracking them.

Reply:
A special format we was for things that were longer term TODOs like for when we do smithy codegen. This one I'm actively working on now, so no other special format.

Reply:
I guess we'll find any remaining ones, should they exist, when we diff before merging to master